Volume/Supporting/AppDelegate.swift

```diff
@@ -6,21 +6,55 @@
 //  Copyright © 2021 Cornell AppDev. All rights reserved.
 //
 
+import AppDevAnalytics
 import UIKit
 import UserNotifications
 
-class AppDelegate: UIResponder, UIApplicationDelegate {
+class AppDelegate: UIResponder, UIApplicationDelegate, UNUserNotificationCenterDelegate {
+
+    private let notificationIntervalKey = "notificationIntervalKey"
+
     func application(_ application: UIApplication, didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey : Any]? = nil) -> Bool {
+        if let _ = launchOptions?[.remoteNotification] {
+            // app not running in background, user taps notification
+            AppDevAnalytics.log(VolumeEvent.clickNotification.toEvent(.notification, id: "", navigationSource: .pushNotification))
+            UserDefaults.standard.set(Date().timeIntervalSince1970, forKey: notificationIntervalKey)
+        }
         return true
     }
     
     func application(_ application: UIApplication, didRegisterForRemoteNotificationsWithDeviceToken deviceToken: Data) {
         let deviceTokenString = deviceToken.map { String(format: "%02.2hhx", $0) }.joined()
         print("UIApplicationDelegate didRegisterForRemoteNotifications with deviceToken: \(deviceTokenString)")
+        print("this is the device token \(deviceToken.reduce("") { $0 + String(format: "%02.2hhx", $1) })")
```

Comment:
Remove this print statement; the print statement on line 28 does the exact same thing.